packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -48,12 +49,15 @@ module.exports = async function contentfulFetch({
     defaultLocale = _.find(contentfulLocales, { default: true }).code
     locales = contentfulLocales.filter(pluginConfig.get(`localeFilter`))
     if (locales.length === 0) {
-      reporter.panic(
-        `Please check if your localeFilter is configured properly. Locales '${_.join(
-          contentfulLocales.map(item => item.code),
-          `,`
-        )}' were found but were filtered down to none.`
-      )
+      reporter.panic({
+        id: CODES.LocalesMissing,
+        context: {
+          sourceMessage: `Please check if your localeFilter is configured properly. Locales '${_.join(
+            contentfulLocales.map(item => item.code),
+            `,`
+          )}' were found but were filtered down to none.`,
```

Comment:
No need for lodash here

```suggestion
          sourceMessage: `Please check if your localeFilter is configured properly. Locales '${
            contentfulLocales.map(item => item.code).join(`,`)
          }' were found but were filtered down to none.`,
```